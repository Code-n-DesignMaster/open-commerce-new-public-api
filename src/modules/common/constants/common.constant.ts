export const VIEWER_FACTORY_TOKEN = 'VIEWER_FACTORY_TOKEN';
export const ROLLBAR_TOKEN = 'ROLLBAR_TOKEN';
