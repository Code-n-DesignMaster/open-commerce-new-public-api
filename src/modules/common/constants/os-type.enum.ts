export enum OS_TYPE {
  ANDROID = 'ANDROID',
  IOS = 'IOS',
}
