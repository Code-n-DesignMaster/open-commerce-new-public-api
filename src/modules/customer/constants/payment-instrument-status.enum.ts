export enum PAYMENT_INSTRUMENT_STATUS {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  PENDING = 'PENDING',
}
