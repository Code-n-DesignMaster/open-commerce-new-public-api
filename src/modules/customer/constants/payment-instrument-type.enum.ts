export enum PAYMENT_INSTRUMENT_TYPE {
  AMAZON_PAY = 'AMAZON_PAY',
  AMERICAN_EXPRESS = 'AMERICAN_EXPRESS',
  APPLE_PAY = 'APPLE_PAY',
  APPLE_PAY_VISA = 'APPLE_PAY_VISA',
  APPLE_PAY_MASTERCARD = 'APPLE_PAY_MASTERCARD',
  APPLE_PAY_AMERICAN_EXPRESS = 'APPLE_PAY_AMERICAN_EXPRESS',
  APPLE_PAY_DISCOVER = 'APPLE_PAY_DISCOVER',
  CARTES_BANCAIRES = 'CARTES_BANCAIRES',
  CARTE_BLANCHE = 'CARTE_BLANCHE',
  CHASE_PAY = 'CHASE_PAY',
  CHINA_UNION_PAY = 'CHINA_UNION_PAY',
  CITI_PAY = 'CITI_PAY',
  CREDIT = 'CREDIT',
  DEBIT = 'DEBIT',
  DINERS_CLUB = 'DINERS_CLUB',
  DISCOVER = 'DISCOVER',
  EFTPOS = 'EFTPOS',
  ELO = 'ELO',
  GIFT = 'GIFT',
  GOOGLE_WALLET = 'GOOGLE_WALLET',
  GOOGLE_PAY = 'GOOGLE_PAY',
  GOOGLE_PAY_VISA = 'GOOGLE_PAY_VISA',
  GOOGLE_PAY_MASTERCARD = 'GOOGLE_PAY_MASTERCARD',
  GOOGLE_PAY_AMERICAN_EXPRESS = 'GOOGLE_PAY_AMERICAN_EXPRESS',
  GOOGLE_PAY_DISCOVER = 'GOOGLE_PAY_DISCOVER',
  ID_CREDIT = 'ID_CREDIT',
  INTERAC = 'INTERAC',
  JCB = 'JCB',
  LASER = 'LASER',
  MAESTRO = 'MAESTRO',
  MASTERCARD = 'MASTERCARD',
  MASTERPASS = 'MASTERPASS',
  PAYPAL = 'PAYPAL',
  PREPAID = 'PREPAID',
  PRIVATE_LABEL = 'PRIVATE_LABEL',
  QUIC_PAY = 'QUIC_PAY',
  SOLO = 'SOLO',
  SUICA = 'SUICA',
  SWITCH = 'SWITCH',
  UNKNOWN = 'UNKNOWN',
  VENMO = 'VENMO',
  VISA = 'VISA',
  VISA_CHECKOUT = 'VISA_CHECKOUT',
  VISA_VPAY = 'VISA_VPAY',
}
