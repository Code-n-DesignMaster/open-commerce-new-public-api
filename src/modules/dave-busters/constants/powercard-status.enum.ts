export enum POWERCARD_STATUS_TYPE {
  DISABLED = 'DISABLED',
  INACTIVE = 'INACTIVE',
  OPEN = 'OPEN',
  STOLEN = 'STOLEN',
  VIP = 'VIP',
  VOID = 'VOID',
}
