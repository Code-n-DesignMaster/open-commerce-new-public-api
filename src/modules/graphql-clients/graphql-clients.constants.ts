export const CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const TRANSACTION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'TRANSACTION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN';
export const AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN =
  'AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN';

export const CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const TRANSACTION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'TRANSACTION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN';
export const AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN =
  'AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN';

export const GRAPHQL_SERVICE_DIRECTORY_TOKEN =
  'GRAPHQL_SERVICE_DIRECTORY_TOKEN';
