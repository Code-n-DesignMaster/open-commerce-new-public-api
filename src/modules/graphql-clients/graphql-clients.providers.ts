import { Provider } from '@nestjs/common';
import {
  CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN,
  NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
  MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN,
  CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN,
  NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
  MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN,
  DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN,
  DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN,
  API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
  API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
  PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN,
  PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN,
  AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN,
  AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN,
  GRAPHQL_SERVICE_DIRECTORY_TOKEN,
} from './graphql-clients.constants';
import {
  CustomerServiceGraphqlClient,
  NotificationServiceGraphqlClient,
  PassManagementServiceGraphqlClient,
  DaveAndBustersServiceGraphqlClient,
  ApiVerificationServiceGraphqlClient,
  PaymentServiceGraphqlClient,
  AuthServiceGraphqlClient,
  GraphqlServiceDirectory,
  SERVICE_NAME,
} from '@open-commerce/db-internal-service-client';
import { CustomerGraphqlClientService } from './clients/customer-graphql-client.service';
import { NotificationGraphqlClientService } from './clients/notification-graphql-client.service';
import { MobilePassGraphqlClientService } from './clients/mobile-pass-graphql-client.service';
import { DaveAndBustersGraphqlClientService } from './clients/dave-and-busters-graphql-client.service';
import { ApiVerificationGraphqlClientService } from './clients/api-verification-graphql-client.service';
import { PaymentGraphqlClientService } from './clients/payment-graphql-client.service';
import { AuthGraphqlClientService } from './clients/auth-graphql-client.service';

export const graphqlClientsProviders: Provider[] = [
  {
    provide: GRAPHQL_SERVICE_DIRECTORY_TOKEN,
    useFactory: () =>
      new GraphqlServiceDirectory([
        SERVICE_NAME.PASS_MANAGEMENT,
        SERVICE_NAME.CUSTOMER,
      ]),
  },
  // NOTE: These are the internal service client instances from
  // internal service client NPM.
  {
    provide: CUSTOMER_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: CustomerServiceGraphqlClient,
  },
  {
    provide: NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: NotificationServiceGraphqlClient,
  },
  {
    provide: MOBILE_PASS_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: PassManagementServiceGraphqlClient,
  },
  {
    provide: DAVE_AND_BUSTERS_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: DaveAndBustersServiceGraphqlClient,
  },
  {
    provide: API_VERIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: ApiVerificationServiceGraphqlClient,
  },
  {
    provide: PAYMENT_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: PaymentServiceGraphqlClient,
  },
  {
    provide: AUTH_SERVICE_GRAPHQL_CLIENT_TOKEN,
    useClass: AuthServiceGraphqlClient,
  },
  //////////////////////////////////////////////

  // NOTE: These are helper abstractions that contain code to
  // build the operations executed by the internal service clients above.
  {
    provide: CUSTOMER_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: CustomerGraphqlClientService,
  },
  {
    provide: NOTIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: NotificationGraphqlClientService,
  },
  {
    provide: MOBILE_PASS_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: MobilePassGraphqlClientService,
  },
  {
    provide: DAVE_AND_BUSTERS_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: DaveAndBustersGraphqlClientService,
  },
  {
    provide: API_VERIFICATION_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: ApiVerificationGraphqlClientService,
  },
  {
    provide: PAYMENT_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: PaymentGraphqlClientService,
  },
  {
    provide: AUTH_GRAPHQL_CLIENT_SERVICE_TOKEN,
    useClass: AuthGraphqlClientService,
  },
  ///////////////////////////////////
];
