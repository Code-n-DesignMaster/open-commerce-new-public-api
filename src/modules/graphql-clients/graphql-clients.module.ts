import { Module, Global } from '@nestjs/common';
import { graphqlClientsProviders } from './graphql-clients.providers';

@Global()
@Module({
  providers: graphqlClientsProviders,
  exports: graphqlClientsProviders,
})
export class GraphqlClientsModule {}
