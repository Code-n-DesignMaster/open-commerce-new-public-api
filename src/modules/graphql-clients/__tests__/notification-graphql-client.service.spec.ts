import { NotificationGraphqlClientService } from '../clients/notification-graphql-client.service';
import { Test } from '@nestjs/testing';
import { NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN } from '../graphql-clients.constants';
import { LoggerService } from '@open-commerce/nestjs-logger';
import { mockGraphqlClientExecuteQuery } from './mock-graphql-client-execute-query';

const mockLogger = {
  debug: jest.fn(),
  warn: jest.fn(),
  info: jest.fn(),
  error: jest.fn(),
  log: jest.fn(),
};

const mockGraphqlClient = {
  executeQuery: jest.fn(),
};

describe('NotificationGraphqlClientService', () => {
  let service: NotificationGraphqlClientService;

  beforeAll(async () => {
    const module = await Test.createTestingModule({
      providers: [
        NotificationGraphqlClientService,
        {
          provide: LoggerService,
          useValue: mockLogger,
        },
        {
          provide: NOTIFICATION_SERVICE_GRAPHQL_CLIENT_TOKEN,
          useValue: mockGraphqlClient,
        },
      ],
    }).compile();

    service = module.get(NotificationGraphqlClientService);
  });

  describe('iosAppNotificationTokenRegister', () => {
    const testCustomerUuid = 'testCustomerUuid';
    const testRawIosToken = 'testRawIosToken';

    beforeAll(() => {
      mockGraphqlClientExecuteQuery(
        'notificationTokenRegister',
        mockGraphqlClient,
      );
    });
    it('should match snapshot', async () => {
      await service.iosAppNotificationTokenRegister(
        testRawIosToken,
        testCustomerUuid,
      );
    });
  });
});
