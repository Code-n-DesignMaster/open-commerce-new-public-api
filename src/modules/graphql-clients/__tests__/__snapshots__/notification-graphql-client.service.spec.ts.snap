// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationGraphqlClientService iosAppNotificationTokenRegister should match snapshot: for operation 1`] = `
"mutation notificationTokenRegister($rawToken: String!, $customerUuid: ID!, $receiverType: NOTIFICATION_RECEIVER_TYPE!) {
 notificationTokenRegister(rawToken: $rawToken, customerUuid: $customerUuid, receiverType: $receiverType) {

  uuid
  token
  customerUuid
  receiverType
 }
}"
`;

exports[`NotificationGraphqlClientService iosAppNotificationTokenRegister should match snapshot: for parameters 1`] = `
Object {
  "customerUuid": "testCustomerUuid",
  "rawToken": "testRawIosToken",
  "receiverType": "IOS_APP",
}
`;
